plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -189,7 +215,7 @@ private Stream<DeltaLakeSplit> getSplits(
         List<DeltaLakeColumnMetadata> predicatedColumns = schema.stream()
                 .filter(column -> predicatedColumnNames.contains(column.getName()))
                 .collect(toImmutableList());
-        return validDataFiles.stream()
```

Comment:
@findepi  I didnt' move the stream logic from `DeltaLakeSplitManager` towards `DeltaLakeSplitSource` because I've noticed (along the way) that this would actually mean moving most of the code of  this class towards `DeltaLakeSplitSource`.

The current state of the code should however not block at all.